### PR TITLE
20403-Clean-class-dependencies-of-Kernel-Tests

### PR DESCRIPTION
**Pull request by pavel-krivanek**
- https://pharo.fogbugz.com/f/cases/20403/

- faltten DatePrintFormatTester
- flatten TDateTimeTest
- move DelayExperimentalCourageousScheduler DelayExperimentalSemaphoreScheduler and DelaySchedulerBackgroundWorkerMorph to Morphic-Tests(-Processes). We had a Morph in Kernel-Tests
- remove RunArray from SelfEvaluatingObjectTest, Tested properly in RunArrayTest
- move MethodDictionaryTest to Collections-Tests (because it uses Collections-Tests traits and so on)
- move MethodPragmaTest to OpalCompiler-Tests, fully dependent on compiler
- move Trait1 isUsed test to TraitTest
- test aliassed method origin on a proper place
- replace DiskStore with ObjectLayout in testHasAbstractMethods, it has the same properties
- make testReferencedClasses more general
- move testComposedBy on a proper place
- move logic of ClassTest testRenaming to ChangeSetClassChangesTest

### CODE REVIEW

**Comment by pavel-krivanek** [comment]
validation result not published but available:
https://ci.inria.fr/pharo-ci-jenkins2/job/Test%20pending%20pull%20request%20and%20branch%20Pipeline/view/change-requests/job/PR-266/

only known testHashMethodNeedsToBeInComparingProtocol is failing

File: src/Kernel-Tests.package/ClassTest.class/instance/testReferencedClasses.st [hunk 1]
```diff
@@ -1,15 +1,9 @@
...
-	{(RBLintRuleTest
-		->
-			{RBRefactoryTestDataApp.
-			Object.
-			Array.
-			WriteStream.
-			String}). 
-	 (RBBasicLintRuleTest -> {RBSelectorEnvironment}).
-	 (RBCompositeLintRuleTest -> {OrderedCollection})
+	{(ExceptionTester -> { MyTestNotification. Warning. String. MyResumableTestError. OrderedCollection. MyTestError}).
```
Comment (guillep) [inline src/Kernel-Tests.package/ClassTest.class/instance/testReferencedClasses.st]: I get your change, and It's good enough. But we should think at some point to make the test only depend on an artificial case.

File: src/Kernel-Tests.package/DatePrintFormatTester.class/instance/assert..st [hunk 2]
```diff
@@ -0,0 +1,5 @@
+asserting
+assert: aBoolean
+	"We check for true explicitly to make the assertion fail for non booleans"
+	aBoolean == true ifFalse: [self signalFailure: 'Assertion failed']
+
```
Comment (guillep) [inline src/Kernel-Tests.package/DatePrintFormatTester.class/instance/assert..st]: All these methods are flattened ones, are they? If so, this is ok.

Comment (pavel-krivanek) [reply src/Kernel-Tests.package/DatePrintFormatTester.class/instance/assert..st]: yes

**Comment by MarcusDenker** [comment]
the failing tests are not related. I will merge

**Pull request merged**